cdap-docs/hydrator-manual/source/developing-plugins/creating-a-plugin.rst

```diff
@@ -63,12 +63,13 @@ the types, only one method is required to be implemented::
 
 .. rubric:: Methods
 
-- ``prepareRun()``: Used to configure the input for each run of the pipeline. This is called by
-  the client that will submit the job for the pipeline run.
+- ``prepareRun()``: Used to configure the input for each run of the pipeline.
+  If the fieldName for a stream or dataset is a macro, their creation will happen during this stage. 
+  This is called by the client that will submit the job for the pipeline run.
 - ``onRunFinish()``: Used to run any required logic at the end of a pipeline run. This is called
   by the client that submitted the job for the pipeline run.
-- ``configurePipeline()``: Used to create any streams or datasets or perform any validation
-  on the application configuration that are required by this plugin.
+- ``configurePipeline()``: Used to perform any validation on the application configuration that is required
+  by this plugin or create any streams or datasets if the fieldName for a stream or dataset is not a macro.
 - ``initialize()``: Initialize the Batch Source. Guaranteed to be executed before any call
```

Comment:
these methods are duplicated for multiple type of plugins, i just updated it for batch-source, though its applicable for all the types.